### ISSUE TITLE
Right-click on commit not selected - actions perform on selected commit instead

### ISSUE THREAD

**Issue by ADTC** [gitx/gitx#406]
Imagine this scenario:

<img width="552" alt="image" src="https://github.com/gitx/gitx/assets/6047296/b9c23072-c696-45d1-9e73-7b0828e5dd58">

---

I have right-clicked (i.e. Control-click) the commit _"Crop images"_ and the menu appears on this commit. Visually, it really looks like the menu is for this commit (with the blue border). And that's what I expect every time I use the menu.

But I'm so frustrated that the actions actually perform on the selected commit _"Github CI"_ (shaded blue). I mean, visually it makes no sense at all.

For example, "Copy Short SHA" will copy `2c5478c` instead of the expected `5ab20a3`. This can be pretty dangerous when pasting these SHA into Terminal, as I often end up acting on the wrong commit (for example cherry-picking it, reverting it or rebasing onto it).

I suggest doing a fix to alter the behavior, to target the right-clicked commit instead of the selected one.

**Comment by SwissalpS** [comment]
I agree, select on rightclick then show contextmenu but only if target is not already selected. (don't deselect single or multiple selections)

**Comment by hannesa2** [comment]
Additional it shows the 10 character of commit message, to avoid confusion https://github.com/gitx/gitx/pull/625

### previous:
<img width="671" height="317" alt="image" src="https://github.com/user-attachments/assets/2ec2a12b-69bb-4d81-bc6c-4a8d00753aa3" />

### new:
<img width="708" height="402" alt="image" src="https://github.com/user-attachments/assets/5c3a12b7-a97c-43ab-9abb-3840be3a06f4" />

**Comment by hannesa2** [comment]
You'll find it here https://github.com/gitx/gitx/releases/tag/1.5.8
